vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/test/java/com/vaadin/flow/component/combobox/ComboBoxTest.java

```diff
@@ -307,6 +306,16 @@ public void setValueWithoutItems_throw() {
         combo.setValue("foo");
     }
 
+    // https://github.com/vaadin/vaadin-flow-components/issues/391
+    @Test
+    public void setValueWithLazyItems_doesntThrow() {
+        final ComboBox<String> comboBox = new ComboBox<>();
+        comboBox.setItems(query -> Stream.of("foo", "bar"));
```

Comment:
As now we have a `setValue` tests for unknown size and in-memory, is that reasonable to add a similar test for lazy loading with defined size?

Another point is to check that if `setValue` is invoked with "absent-item", it won't throw anything if the items are set to combobox, so let's add a corresponding test.

Reply:
>As now we have a setValue tests for unknown size and in-memory, is that reasonable to add a similar test for lazy loading with defined size?

I don't see how the defined/undefined size makes any difference to the `setValue` code, for completeness sake I can do it if there are other changes required

>Another point is to check that if setValue is invoked with "absent-item", it won't throw anything if the items are set to combobox, so let's add a corresponding test.

To be clear, you're referring to the case where the set item is not fetched on the first range ? Or what ? The item is not at all available ?

Reply:
> The item is not at all available ?

Yes, I consider this case.

Reply:
Just to verify that no exception is thrown when the the item "baz" is not present

Reply:
>Just to verify that no exception is thrown when the the item "baz" is not present

I tried to add what I'd consider a meaningful test case for this, but failed because I cannot figure out a nice way to set the request range for the ComboBox, which is needed if I want to get a fetch triggered. Otherwise it I don't think it is meaningful. I don't want to spend any more time on this as it is irrelevant for the bug in question. Someone (else) can do that in another PR.